### PR TITLE
Crawler to be run for promedmail

### PR DESCRIPTION
**Pull request by aishwaryakapse**
Enter the path of the config file as a command line parameter to start running the crawler. This is storing the crawled files in a local file system.

Although the crawler had been configured for both proquest and promedmail, proquest is disallowing bots to search.

I have removed the config file for proquest as it was of no use for now.

I have removed the configuration for setting the maximum running time of the crawler. Please let me know if that has to be set to a definite amount of time. Otherwise the crawler runs for indefinite amount of time.

Will need some changes in the CrawlerDriver if we want to make it more general and user-friendly. This is because we need to make add click/dont-click elements depending on the domain we are crawling.

#66 
@JavierJia @xuxip @MoniMoledo @dennis09121111

### CODE REVIEW

File: noah/src/main/resources/crawler/promedmail.conf [hunk 1]
```diff
@@ -5,7 +5,7 @@
 crawljax.url = "http://www.promedmail.org"
 
 #The name of the folder in which states are stored
-crawljax.outputDir = "outputfolder"
+crawljax.outputDir = "outputfolderpromedmail"
```
Comment (JavierJia) [inline noah/src/main/resources/crawler/promedmail.conf]: use camel case or `_` to make the folder readable.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala [hunk 2]
```diff
@@ -6,84 +6,134 @@ import java.util.concurrent.TimeUnit
...
+    val builder = CrawljaxConfiguration.builderFor(EntryURL)
 
-  builder.crawlRules.addCrawlCondition("stay within promedmail domain", new UrlCondition(EntryURL+"/"))
+    builder.crawlRules.addCrawlCondition("stay within " + EntryURL + "domain", new UrlCondition(EntryURL + "/"))
 
-  builder.setOutputDirectory(new File(OutputDir))
+    builder.setOutputDirectory(new File(OutputDir))
 
-  val browser = BrowserType.FIREFOX
+    builder.crawlRules().clickOnce(true)
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala]: what does this line do?

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala [hunk 3]
```diff
@@ -6,84 +6,134 @@ import java.util.concurrent.TimeUnit
...
+      builder.crawlRules().setInputSpec(in)
+    }
 
-  while(it.hasNext) {
-    builder.crawlRules().dontClick("table").withAttribute("class",it.next())
-  }
+    /* Get the dont click elements to be configured */
+    /* DIV */
+    if(conf.hasPath("builder.crawlrules.dontclickelements")) {
+      val DivIds = conf.getStringList("builder.crawlrules.dontclickelements.div.id")
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala]: these lines look very similar, could you rewrite them in a more concise way?

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala [hunk 4]
```diff
@@ -6,84 +6,134 @@ import java.util.concurrent.TimeUnit
...
+        builder.crawlRules().click("a").withAttribute("id", it.next())
+      }
+
+      it = AClasses.iterator()
+      while(it.hasNext) {
+        builder.crawlRules().click("a").withAttribute("class", it.next())
+      }
+    }
+
+    /*Crawloverview plugin generates an HTML report with a snapshot overview of what is crawled by Crawljax.
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala]: Good doc!

File: noah/src/main/resources/crawler/promedmail.conf (hunk 1 above)
Comment (aishwaryakapse) [inline noah/src/main/resources/crawler/promedmail.conf]: done

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala (hunk 2 above)
Comment (aishwaryakapse) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala]: It was from previous commit. Has been removed in the new one.

Done.

File: noah/src/main/resources/crawler/promedmail.conf [hunk 5]
```diff
@@ -16,27 +16,22 @@ crawljax.waitafterevent = 50
...
-      }
-
-      span {
-        class = ["printable"]
+    clickelements {
+      a {
+        id = []
+        href = ["http://promedmail.org"]
+        class = ["lcl"]
+        text = ["« prev"]
```
Comment (JavierJia) [inline noah/src/main/resources/crawler/promedmail.conf]: did you see this wield `<<`?

Comment (aishwaryakapse) [inline noah/src/main/resources/crawler/promedmail.conf]: Yes.... It works to my knowledge. Would you like to see it work? It does not take much time.

**Comment by JavierJia** [comment]
👍

**Pull request merged**